t/lib/Helper.pm

```diff
@@ -2,18 +2,47 @@ package Helper;
 
 use strict;
 use warnings;
-use Exporter ();
+
+use parent 'Exporter';
+use Test::More;
+
 use PPI::Document ();
 
-our @ISA = "Exporter";
-our @EXPORT_OK = qw( check_with );
+our @EXPORT_OK = qw( check_with  safe_new );
+
+=head1 safe_new @args
+
+	my $doc = safe_new \"use strict";
+
+Creates a PPI::Document object from the arguments and reports errors if
+necessary. Can be used to replace most document new calls in the tests for
+easier testing.
+
+=cut
+
+sub safe_new {
+	my $Document = PPI::Document->new(@_);
+	is( PPI::Document->errstr, undef ) if PPI::Document->errstr;
```

Comment:
Why not just run this test under every condition? Won't it mess up the hardcoded test plans in the cases where it is executed (and will always fail)?

Reply:
it's a forced fail, but using is() there makes the output more readable, and since it's already a fail, the test count doesn't matter either. i will add a comment tho :)

Reply:
hm, maybe just using is is actually ok?

Reply:
yup, changed that out, and i think this bit is ok to resolve :)